src/OpenLoco/Objects/WallObject.cpp

```diff
@@ -1,9 +1,27 @@
 #include "WallObject.h"
 #include "../Graphics/Colour.h"
 #include "../Graphics/Gfx.h"
+#include "../Interop/Interop.hpp"
 
 namespace OpenLoco
 {
+    // 0x004C4ACA
+    void WallObject::load(const LoadedObjectHandle handle, stdx::span<std::byte> data)
+    {
+        Interop::registers regs;
+        regs.esi = Interop::X86Pointer(this);
+        regs.ebx = handle.id;
+        regs.ecx = enumValue(handle.type);
+        Interop::call(0x004C4ACA, regs);
+    }
+
+    // 0x004CAF0
+    void WallObject::unload()
+    {
+        name = 0;
+        sprite = 0;
```

Comment:
The unload must only clear things set by load. (Otherwise you corrupt the object as we learnt in openrct2). Unfortunately this pr only shows the unload so apart from assembly there is nothing to check against

Reply:
> The unload must only clear things set by load.

That would imply that state is being shared between objects? But yeah fair enough